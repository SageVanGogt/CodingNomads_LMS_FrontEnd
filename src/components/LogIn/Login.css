.LogIn {
  border: 1px solid #0066B4;
  width: 300px;
  height: 150px;
  margin: auto;
  position: relative;
  top: 200px;
  left: 100px; }

.LogIn_tab {
  box-sizing: border-box;
  width: 50%;
  border-bottom: 1px solid #0066B430;
  display: inline-block;
  height: 25px;
  padding-top: 3px;
  cursor: pointer; }

.LogIn_tab__active {
  background-color: #0066B430; }

.LogIn input {
  width: 70%;
  height: 20px;
  margin: 10px auto;
  display: block;
  padding-left: 5px; }

.LogIn button {
  width: 70%;
  height: 25px;
  border: .5px solid #0066B430; }

.LogIn button:enabled {
  background-color: #0066B430; }
