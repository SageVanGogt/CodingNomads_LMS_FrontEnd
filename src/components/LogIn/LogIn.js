import React, { Component} from 'react';
import { connect } from 'react-redux';
import './Login.css';
import * as API from '../../apiCalls/apiCalls';

export class LogIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      userbase: 'student',
      email: '',
      password: ''
    };
  }

  setUserbase = (userbase) => {
    this.setState({userbase});
  }

  onSubmit = async (event) => {
    event.preventDefault();
    const creds = {
      email: this.state.email, 
      password: this.state.password
    }
    try {
      const response = await API.authUser(creds);
      const user = {
        id: response.id,
        name: response.firstName + response.lastName
      }
      localStorage.setItem('id_token', user.id_token)
      localStorage.setItem('id_token', user.access_token)
    } catch (error) {
      throw error;
    }

    this.setState({
      email: '',
      password: ''
    });
  }

  render() {
    const tabs = this.state.userbase === 'student' ? 
      <div className='LogIn_tabs'>
        <div 
          className='LogIn_tab LogIn_tab__active' 
          onClick={() => this.setUserbase('student')}
        >
          <span>Student</span>
        </div>
        <div 
          className='LogIn_tab' 
          onClick={() => this.setUserbase('admin')}
        >
          <span>Admin</span>
        </div>
      </div>
      :
      <div className='LogIn_tabs'>
        <div 
          className='LogIn_tab' 
          onClick={() => this.setUserbase('student')}
        >
          <span>Student</span>
        </div>
        <div 
          className='LogIn_tab LogIn_tab__active'
          onClick={() => this.setUserbase('admin')}
        >
          <span>Admin</span>
        </div>
      </div>;

    const { email, password } = this.state;

    const isInvalid = password === '' || email === '';

    return (
      <div className='LogIn'>
        {tabs}
        <form onSubmit={this.onSubmit} className='signIn authentication'>
          <input
            className='LogIn__email'
            value={email}
            onChange={event => this.setState({email: event.target.value})}
            type="text"
            placeholder="Email Address"
          />
          <input
            className='LogIn__password'
            value={password}
            onChange={event => this.setState({ password: event.target.value})}
            type="password"
            placeholder="Password"
          />
          <button disabled={isInvalid} type="submit">
            Sign In
          </button>
        </form>
      </div>
    );
  }
}

export default connect(null, null)(LogIn);