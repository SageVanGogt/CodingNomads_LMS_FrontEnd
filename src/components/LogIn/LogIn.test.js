import React from 'react';
import { shallow } from 'enzyme';
import { LogIn } from './LogIn';

describe('LogIn', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LogIn/>);
  });

  it('should match snapshot with default state', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with userbase of admin in state', () => {
    wrapper.setState({userbase: 'admin'});

    expect(wrapper).toMatchSnapshot();
  });
});