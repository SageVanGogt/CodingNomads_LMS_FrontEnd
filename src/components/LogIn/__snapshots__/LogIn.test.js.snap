// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should match snapshot with default state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="LogIn_tabs"
>
          <div
                    className="LogIn_tab LogIn_tab__active"
                    onClick={[Function]}
          >
                    <span>
                              Student
                    </span>
          </div>
          <div
                    className="LogIn_tab"
                    onClick={[Function]}
          >
                    <span>
                              Admin
                    </span>
          </div>
</div>,
        <form
          className="signIn authentication"
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          <input
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign In
          </button>
</form>,
      ],
      "className": "LogIn",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="LogIn_tab LogIn_tab__active"
              onClick={[Function]}
>
              <span>
                            Student
              </span>
</div>,
            <div
              className="LogIn_tab"
              onClick={[Function]}
>
              <span>
                            Admin
              </span>
</div>,
          ],
          "className": "LogIn_tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Student
</span>,
              "className": "LogIn_tab LogIn_tab__active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Student",
              },
              "ref": null,
              "rendered": "Student",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Admin
</span>,
              "className": "LogIn_tab",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Admin",
              },
              "ref": null,
              "rendered": "Admin",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            <input
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign In
</button>,
          ],
          "className": "signIn authentication",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="LogIn_tabs"
>
            <div
                        className="LogIn_tab LogIn_tab__active"
                        onClick={[Function]}
            >
                        <span>
                                    Student
                        </span>
            </div>
            <div
                        className="LogIn_tab"
                        onClick={[Function]}
            >
                        <span>
                                    Admin
                        </span>
            </div>
</div>,
          <form
            className="signIn authentication"
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="text"
                        value=""
            />
            <input
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign In
            </button>
</form>,
        ],
        "className": "LogIn",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="LogIn_tab LogIn_tab__active"
                onClick={[Function]}
>
                <span>
                                Student
                </span>
</div>,
              <div
                className="LogIn_tab"
                onClick={[Function]}
>
                <span>
                                Admin
                </span>
</div>,
            ],
            "className": "LogIn_tabs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Student
</span>,
                "className": "LogIn_tab LogIn_tab__active",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Student",
                },
                "ref": null,
                "rendered": "Student",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Admin
</span>,
                "className": "LogIn_tab",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Admin",
                },
                "ref": null,
                "rendered": "Admin",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
/>,
              <input
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign In
</button>,
            ],
            "className": "signIn authentication",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LogIn should match snapshot with userbase of admin in state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="LogIn_tabs"
>
          <div
                    className="LogIn_tab"
                    onClick={[Function]}
          >
                    <span>
                              Student
                    </span>
          </div>
          <div
                    className="LogIn_tab LogIn_tab__active"
                    onClick={[Function]}
          >
                    <span>
                              Admin
                    </span>
          </div>
</div>,
        <form
          className="signIn authentication"
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          <input
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign In
          </button>
</form>,
      ],
      "className": "LogIn",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="LogIn_tab"
              onClick={[Function]}
>
              <span>
                            Student
              </span>
</div>,
            <div
              className="LogIn_tab LogIn_tab__active"
              onClick={[Function]}
>
              <span>
                            Admin
              </span>
</div>,
          ],
          "className": "LogIn_tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Student
</span>,
              "className": "LogIn_tab",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Student",
              },
              "ref": null,
              "rendered": "Student",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Admin
</span>,
              "className": "LogIn_tab LogIn_tab__active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Admin",
              },
              "ref": null,
              "rendered": "Admin",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            <input
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign In
</button>,
          ],
          "className": "signIn authentication",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="LogIn_tabs"
>
            <div
                        className="LogIn_tab"
                        onClick={[Function]}
            >
                        <span>
                                    Student
                        </span>
            </div>
            <div
                        className="LogIn_tab LogIn_tab__active"
                        onClick={[Function]}
            >
                        <span>
                                    Admin
                        </span>
            </div>
</div>,
          <form
            className="signIn authentication"
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="text"
                        value=""
            />
            <input
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign In
            </button>
</form>,
        ],
        "className": "LogIn",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="LogIn_tab"
                onClick={[Function]}
>
                <span>
                                Student
                </span>
</div>,
              <div
                className="LogIn_tab LogIn_tab__active"
                onClick={[Function]}
>
                <span>
                                Admin
                </span>
</div>,
            ],
            "className": "LogIn_tabs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Student
</span>,
                "className": "LogIn_tab",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Student",
                },
                "ref": null,
                "rendered": "Student",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Admin
</span>,
                "className": "LogIn_tab LogIn_tab__active",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Admin",
                },
                "ref": null,
                "rendered": "Admin",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
/>,
              <input
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign In
</button>,
            ],
            "className": "signIn authentication",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
